### DISCUSSION TITLE
Data importer using GoCardless should ignore "Pending Transactions"

### DISCUSSION THREAD

**Discussion by matt-gretton-dann** [firefly-iii/firefly-iii#8783] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)
- [X] My request is not listed as [a very good idea, but unfortunately...](https://docs.firefly-iii.org/explanation/more-information/what-its-not/)
- [X] I've used [the search](https://github.com/firefly-iii/firefly-iii/issues?q=is%3Aissue) and this has not been requested before.

### Description

When using the Data Importer to import data via GoCardless for my UK Bank (Nationwide) I am getting pending transactions included in the import.  These are transactions that have not yet cleared through the banks system and may include amounts which are never cleared because the originator changes their mind (hotels, taxis do this sort of thing).

This is an issue because some of these pending transactions may never end up being real transactions.  And when they do end up as real transactions they have a different date to the pending transaction's date, making reconciliation hard.

I cannot filter by date on the import as the pending transactions may have the same date as real transactions.

### Solution

I would like an option in the data importer to ignore pending transactions.

### What are alternatives?

Manually removing pending transactions after import.

### Additional context

Pending transactions have previously come up in:
 - #5300 where the requestor wanted them to go to a separate account.  The request here is different, I want them not to be imported.
 - #3241 where the requestor had the inverse problem with the Spectre API: Pending transactions not appearing.

My request is different from both the above in that it asks for data that is available to not be imported.

Looking at the GoCardless API it should be possible to determine which transactions are pending.

**Comment by JC5** [comment]
Thanks for the suggestion. This behavior will not change.

**Comment by JC5** [comment]
If you create a rule that responds to the tag "pending", and then deletes the offending transactions, they will be removed automatically.

**Comment by mciantar** [comment]
I am having the same issue but it is not consistent. I am using GoCardless to import feeds from 2 different banks. From one, I am not having this issue, and somehow, the transactions for that bank get updated and the pending entry is removed. For the other, I am getting duplicates. What would be the reason for the different behavior?
